lib/asciidoctor.rb

```diff
@@ -7,10 +7,9 @@
 
 require 'set'
 
-# NOTE RUBY_ENGINE == 'opal' conditional blocks are filtered by the Opal preprocessor
+# NOTE RUBY_ENGINE == 'opal' conditional blocks like this are filtered by the Opal preprocessor
 if RUBY_ENGINE == 'opal'
-  # NOTE asciidoctor/opal_ext is supplied by the Asciidoctor.js build
-  require 'asciidoctor/opal_ext'
+  require 'asciidoctor/opal' # supplies compile and runtime overrides for the Opal environment; fulfilled by Asciidoctor.js
```

Comment:
> It's more technically correct to put the require at the top of the file. I plan to add it right under the require for opal_ext. 

I've done some tests recently and with the require at the top, I couldn't override `path_resolver` methods.

> Then, you can decide on your end what you want that to resolve to. This is a clear API between the two compilers, I think. Would you agree?

`require 'asciidoctor/opal'` is fine :smile: